components/fxa-client/src/events.rs

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::FxaState;
+use crate::Device;
+
+/// FxA event handler trait
+pub trait FxaEventHandler {
+    fn on_event(&self, event: FxaEvent);
+    fn on_command(&self, command: FxaCommand);
+}
+
+/// FxA event that the app may want to respond to
+pub enum FxaEvent {
+    /// The FxA client state changed
+    StateChanged {
+        state: FxaState,
+    },
+    /// The account itself was deleted
+    AccountDeleted,
+    /// Error when trying to migrate old account data
+    AccountMigrationFailed {
+        // TODO: what fields would be useful here
+    },
+    /// The account's profile was updated
+    ProfileUpdated,
+    /// The list of devices changed
+    DevicesChanged {
+        /// Device record for this client
+        client_devices: Device,
+        /// Device record for all other known clients
+        other_devices: Vec<Device>,
+    },
+    // I don't think we need these:
```

Comment:
These are delivered by push, so I think we probably do want them - eg, desktop still displays UI for `DeviceConnected` and IIRC even checks if itself was disconnected in the `DeviceDisconnected` state to leave the browser as though it never signed in.

Reply:
My hope was that `DevicesChanged` could be used a a single event type for all of these, but if that's not going to work out we can definitely add these back in.

Reply:
Oh right - yeah, that might work just fine.